### PR TITLE
Samples-Style Refactor, Minor Fixes and Refactoring

### PR DESCRIPTION
**Pull request by seveibar**
* This breaks amplify, I couldn't understand the code in the `ImportPage` enough to refactor, I also don't think it should be in the `ImportPage`- it should be in the `ImportFromS3Dialog` since it seems very specific to the S3 implementation and not importing samples in general. I'll probably need help to fix the commented TODOs in `ImportFromS3Dialog` @CedricJean 
* This PR will merge without releasing because we're going to layer on some regression tests, hopefully a test for S3 functionality as well

### CODE REVIEW

**Comment by CedricJean** [comment]
The import page code is messy my bad i can explain its general function : it is to deal with the data retrieved from s3. The biggest change is i create an additionnal way to download a project. In fact with our new feature it have two ways to addSample by downloading the project who include the annotation and the simple and traditionnal way were we just add the sample or the data.

And now its particulary messy because the function in import page include both.

File: src/components/ImportFromS3Dialog/get-annotation-from-aws.js [hunk 1]
```diff
@@ -21,21 +22,12 @@ export default async (result, samples, folderToFetch, authConfig) => {
...
-                  ) {
-                    sampleName = json.content.taskData[i].sampleName
-                  } else {
-                    sampleName = getSampleNameFromURL(
-                      json.content.taskData[i]
-                    )[1]
-                  }
+                var newSamples = []
+                for (var i = 0; i < samples.length; i++) {
+                  var sampleName = getSampleNameFromURL(json.content.samples[i])
```
Comment (puskuruk) [inline src/components/ImportFromS3Dialog/get-annotation-from-aws.js:30]: that's good refactor :rocket:

File: src/components/ImportPage/index.js [hunk 2]
```diff
@@ -125,20 +125,6 @@ const Button = ({
   )
 }
 
-const convertToTaskDataObject = (fp) => {
```
Comment (puskuruk) [inline src/components/ImportPage/index.js:128]: That's good refactor. Made code much more cleaner :rocket:

File: src/components/ImportPage/give-sample-name.js [hunk 3]
```diff
@@ -1,50 +0,0 @@
-import getSampleNameFromURL from "../../utils/get-sample-name-from-url"
```
Comment (puskuruk) [inline src/components/ImportPage/give-sample-name.js:1]: :+1:

File: src/components/ImportPage/prompt-and-get-samples-from-local-directory.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+// @flow weak
+
```
Comment (puskuruk) [inline src/components/ImportPage/prompt-and-get-samples-from-local-directory.js:2]: That's a good and readable file :+1:

File: src/components/LocalStorageApp/index.js [hunk 5]
```diff
@@ -82,6 +82,7 @@ export default () => {
       })
   }
 
+  // TODO centralize this with other auth provider code
```
Comment (puskuruk) [inline src/components/LocalStorageApp/index.js:85]: :+1:

File: src/components/LoginDrawer/SignIn.js [hunk 6]
```diff
@@ -29,7 +29,9 @@ export default ({
   onRequireCompleteSignUp,
   onClose,
 }) => {
-  Amplify.configure(authConfig)
+  try {
+    Amplify.configure(authConfig)
+  } catch (e) {}
```
Comment (puskuruk) [inline src/components/LoginDrawer/SignIn.js:34]: That's a good idea :+1:

File: src/components/TextClassification/index.js [hunk 7]
```diff
@@ -5,11 +5,12 @@ import NLPAnnotator from "react-nlp-annotate/components/NLPAnnotator"
...
 export const TextClassification = (props) => {
   const [currentSampleIndex, changeCurrentSampleIndex] = useState(0)
-  const initialLabels =
-    props.taskOutput && props.taskOutput[currentSampleIndex]
-      ? [props.taskOutput[currentSampleIndex].label] ||
-        props.taskOutput[currentSampleIndex].labels
-      : undefined
+  const { annotation } = props.samples[currentSampleIndex]
+  const initialLabels = annotation
+    ? typeof annotation === "string"
```
Comment (puskuruk) [inline src/components/TextClassification/index.js:10]: :+1:

File: src/components/TextClassification/index.story.js [hunk 8]
```diff
@@ -28,8 +28,10 @@ storiesOf("TextClassification", module)
           },
         ],
       }}
-      taskData={[{ document: "Harry" }, { document: "Malfoy" }]}
-      taskOutput={[{ label: "gryffindor" }, null]}
+      samples={[
+        { document: "Harry", annotation: "gryffindor" },
```
Comment (puskuruk) [inline src/components/TextClassification/index.story.js:32]: Ricky will love that :smile:

File: src/vanilla/image-classification-layouts.story.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+// @flow
+
```
Comment (puskuruk) [inline src/vanilla/image-classification-layouts.story.js:2]: That's really good to having a story :tada:

Comment (puskuruk) [inline src/vanilla/image-classification-layouts.story.js:2]: That's really good to having a story :tada:

**Pull request merged**